recurring_contract/models/recurring_contract.py

```diff
@@ -307,7 +325,6 @@ def contract_waiting(self):
             raise UserError(_("Active contract cannot be put to waiting"))
         if self.filtered(lambda c: not c.total_amount):
             raise UserError(_("Please configure contract lines"))
-        self.mapped("group_id").generate_invoices()
```

Comment:
Was it a duplicate call? Will the invoices still be generated when we validate a contract even after this change?

Reply:
It was not a duplicate call. This line cause issue on most of the test_case in the accounting module as they all make direct call to generate_invoice after the sponsorship is set to the 'waiting' state. 
No, they would have to be generated trough the button.

Reply:
Then please restore it as we want that the invoices automatically get generated when we validate the sponsorship. The test cases should be adapted to this behaviour if needed.

Reply:
Actually I was wrong... 

Looking back at the code in accounting I realized that it is not working properly.
The generation appends before the change of state. But invoices are only generated for waiting and active contract.

It was working because, as you suggested, this code was actually duplicated in an other module : https://github.com/CompassionCH/compassion-modules/blob/12.0/sponsorship_compassion/models/contracts.py#L1064 

Should  I left the code as is or do the opposite and handle the generation in accounting ?

Reply:
You can leave it as it is now. Thanks for checking! :+1: